flow: add a `prometheus.exporter.cadvisor` component (#5307)
grafana/agent: docs/sources/flow/reference/components/prometheus.exporter.cadvisor.md

}
```

## Arguments
The following arguments can be used to configure the exporter's behavior.
All arguments are optional. Omitted fields take their default values.

Name | Type | Description | Default | Required
---- | ---- | ----------- | ------- | --------
`store_container_labels` | `bool` | Whether to convert container labels and environment variables into labels on Prometheus metrics for each container. | `true` | no
`allowlisted_container_labels` | `list(string)` | Allowlist of container labels to convert to Prometheus labels. | | no
`env_metadata_allowlist` | `list(string)` | Allowlist of environment variable keys matched with specified prefix that needs to be collected for containers. | | no
`raw_cgroup_prefix_allowlist` | `list(string)` | List of cgroup path prefix that needs to be collected, even when docker_only is specified. | | no
`perf_events_config` | `string` | Path to a JSON file containing configuration of perf events to measure. | | no
`resctrl_interval` | `duration` | Interval to update resctrl mon groups. | `0` | no
`disabled_metrics` | `list(string)` | List of metrics to be disabled which, if set, overrides the default disabled metrics. | | no

> `enabled_metrics` | `list(string)` | List of metrics to be enabled which, if set, overrides disabled_metrics. | | no

Author: AFAICT this list is not about the actual Prometheus metrics being exposed, but rather the `MetricKind` we want to expose (or not). Here's a list of the supported values, I think it's worth documenting them.

https://github.com/grafana/cadvisor/blob/fbd519ba03978d54cb54ea7ed8ab9d6e3dd64590/container/factory.go#L48-L68

- /docs/grafana-cloud/monitor-infrastructure/agent/flow/reference/components/prometheus.exporter.cadvisor/
- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/reference/components/prometheus.exporter.cadvisor/
canonical: https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.exporter.cadvisor/
title: prometheus.exporter.cadvisor
description: Learn about the prometheus.exporter.cadvisor
---

# prometheus.exporter.cadvisor
The `prometheus.exporter.cadvisor` component exposes container metrics using
[cAdvisor](https://github.com/google/cadvisor).

## Usage

```river
prometheus.exporter.cadvisor "LABEL" {
}
```

## Arguments
The following arguments can be used to configure the exporter's behavior.
All arguments are optional. Omitted fields take their default values.

Name | Type | Description | Default | Required
---- | ---- | ----------- | ------- | --------
`store_container_labels` | `bool` | Whether to convert container labels and environment variables into labels on Prometheus metrics for each container. | `true` | no
`allowlisted_container_labels` | `list(string)` | Allowlist of container labels to convert to Prometheus labels. | `[]`  | no

> `env_metadata_allowlist` | `list(string)` | Allowlist of environment variable keys matched with specified prefix that needs to be collected for containers. | `[]` | no

Reviewer: ```suggestion
`env_metadata_allowlist` | `list(string)` | Allowlist of environment variable keys matched with a specified prefix that needs to be collected for containers. | `[]` | no
```

> `raw_cgroup_prefix_allowlist` | `list(string)` | List of cgroup path prefix that needs to be collected, even when docker_only is specified. | `[]` | no

Reviewer: ```suggestion
`raw_cgroup_prefix_allowlist` | `list(string)` | List of cgroup path prefixes that need to be collected, even when docker_only is specified. | `[]` | no
```

> `perf_events_config` | `string` | Path to a JSON file containing configuration of perf events to measure. | `""` | no

Reviewer: ```suggestion
`perf_events_config` | `string` | Path to a JSON file containing the configuration of perf events to measure. | `""` | no
```
